lib/archive/archiver.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"io"
+	"net/url"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	// ExcludeString defines the string that specifies a path exclusion.
+	ExcludeString = "exclude"
+	// IncludeString defines the string that specifies a path inclusion.
+	// Note: An included path that sits under an excluded path should be included.
+	// Example: if / is excluded, and /files is included, /files should  be added to the archive.
+	IncludeString = "include"
+	// RebaseString defines the string that specifies a path rebase.
+	// The rebase path is prepended to the path in the archive header.
+	RebaseString = "rebase"
+)
+
+// PathSpec describes rules for handling specified paths during archival
+type PathSpec struct {
+	Inclusions map[string]struct{}
+	Exclusions map[string]struct{}
+	RebasePath string
+}
+
+// Archiver defines an API for creating archives consisting of data that
+// exist between a child and parent layer, as well as unpacking archives
+// to container filesystems
+type Archiver interface {
+
+	// ReadArchive reads the delta between child and ancestor layers, returning
+	// the difference as a tar archive.
+	//
+	// store - the store containing the two layers
+	// child - the new layer
```

Comment:
The diffspeak I've been reading lately refers to `old` and `new` so that's what I've used in `diff.go`. It just operates on mount points like you said @vburenin. However, this is for `Export` (which actually deals with the disk attach/mounts) before calling into `diff`, so at this level `id` and `ancestor` makes sense.